Two beans with @Scheduled causes jaeger metrics error

There seems to be an issue with the combination of two (or more) beans using @Scheduled with @Traced at the same time. 

Quarkus 0.14
Mvn 3.6.1
Java 11
GraalVM not relevant

Trigger using code sample in this repository:
https://github.com/mejlholm/quarkus-scheduled-error

Though it seems to be an issue with @Traced and @Scheduled in two @ApplicationScoped beans at the same time. 

Triggers this error on startup with: mvn clean compile quarkus:dev

2019-04-29 11:07:17,088 ERROR [org.qua.cor.ErrorLogger] (DefaultQuartzScheduler_Worker-1) Job (io.quarkus.scheduler.Scheduler.1_org.acme.quickstart.Scheduler1_ScheduledInvoker_run_72e66771a77415a7284d3ae42331659c186071de threw an exception.: org.quartz.SchedulerException: Job threw an unhandled exception. [See nested exception: java.lang.RuntimeException: No reflection exceptions should be thrown unless there is a fundamental error in your code set up.]
        at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
        at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: java.lang.RuntimeException: No reflection exceptions should be thrown unless there is a fundamental error in your code set up.
        at io.jaegertracing.internal.metrics.Metrics.createMetrics(Metrics.java:76)
        at io.jaegertracing.internal.metrics.Metrics.<init>(Metrics.java:32)
        at io.jaegertracing.internal.metrics.Metrics.<init>(Metrics.java:28)
        at io.jaegertracing.Configuration.getTracerBuilder(Configuration.java:221)
        at io.jaegertracing.Configuration.getTracer(Configuration.java:245)
        at io.quarkus.jaeger.runtime.QuarkusJaegerTracer.lambda$tracer$0(QuarkusJaegerTracer.java:45)
        at java.base/java.util.concurrent.atomic.AtomicReference.updateAndGet(AtomicReference.java:209)
        at io.quarkus.jaeger.runtime.QuarkusJaegerTracer.tracer(QuarkusJaegerTracer.java:41)
        at io.quarkus.jaeger.runtime.QuarkusJaegerTracer.buildSpan(QuarkusJaegerTracer.java:51)
        at io.opentracing.util.GlobalTracer.buildSpan(GlobalTracer.java:133)
        at io.opentracing.contrib.interceptors.OpenTracingInterceptor.wrap(OpenTracingInterceptor.java:44)
        at io.opentracing.contrib.interceptors.OpenTracingInterceptor_Bean.intercept(Unknown Source)
        at io.quarkus.arc.InvocationContextImpl$InterceptorInvocation.invoke(InvocationContextImpl.java:270)
        at io.quarkus.arc.InvocationContextImpl.invokeNext(InvocationContextImpl.java:149)
        at io.quarkus.arc.InvocationContextImpl.proceed(InvocationContextImpl.java:173)
        at io.quarkus.arc.ActivateRequestContextInterceptor.aroundInvoke(ActivateRequestContextInterceptor.java:38)
        at io.quarkus.arc.ActivateRequestContextInterceptor_Bean.intercept(Unknown Source)
        at io.quarkus.arc.InvocationContextImpl$InterceptorInvocation.invoke(InvocationContextImpl.java:270)
        at io.quarkus.arc.InvocationContextImpl.invokeNext(InvocationContextImpl.java:149)
        at io.quarkus.arc.InvocationContextImpl.proceed(InvocationContextImpl.java:173)
        at org.acme.quickstart.Scheduler1_Subclass.run(Unknown Source)
        at org.acme.quickstart.Scheduler1_ClientProxy.run(Unknown Source)
        at org.acme.quickstart.Scheduler1_ScheduledInvoker_run_72e66771a77415a7284d3ae42331659c186071de.invoke(Unknown Source)
        at io.quarkus.scheduler.runtime.QuartzScheduler$InvokerJob.execute(QuartzScheduler.java:264)
        at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
        ... 1 more
Caused by: java.lang.IllegalArgumentException: A metric with name jaeger_tracer_reporter_queue_length already exists
        at io.smallrye.metrics.MetricsRegistryImphttps://github.com/quarkusio/quarkus/issues/new

l.register(MetricsRegistryImpl.java:113)
        at io.quarkus.jaeger.runtime.QuarkusJaegerMetricsFactory.createGauge(QuarkusJaegerMetricsFactory.java:64)
        at io.jaegertracing.internal.metrics.Metrics.createMetrics(Metrics.java:66)
        ... 25 more

I was able to reproduce with `0.14.0` but not with `999-SNAPSHOT`. It might have been fixed in https://github.com/quarkusio/quarkus/pull/2156

```
777d19e98d20799c6b72c29b26289f3f771250 threw an exception.: org.quartz.SchedulerException: Job threw an unhandled exception. [See nested exception: java.lang.RuntimeException: No reflection exceptions should be thrown unless there is a fundamental error in your code set up.]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: java.lang.RuntimeException: No reflection exceptions should be thrown unless there is a fundamental error in your code set up.
	at io.jaegertracing.internal.metrics.Metrics.createMetrics(Metrics.java:76)
	at io.jaegertracing.internal.metrics.Metrics.<init>(Metrics.java:32)
	at io.jaegertracing.internal.metrics.Metrics.<init>(Metrics.java:28)
	at io.jaegertracing.Configuration.getTracerBuilder(Configuration.java:221)
	at io.jaegertracing.Configuration.getTracer(Configuration.java:245)
	at io.quarkus.jaeger.runtime.QuarkusJaegerTracer.lambda$tracer$0(QuarkusJaegerTracer.java:45)
	at java.util.concurrent.atomic.AtomicReference.updateAndGet(AtomicReference.java:179)
	at io.quarkus.jaeger.runtime.QuarkusJaegerTracer.tracer(QuarkusJaegerTracer.java:41)
	at io.quarkus.jaeger.runtime.QuarkusJaegerTracer.buildSpan(QuarkusJaegerTracer.java:51)
	at io.opentracing.util.GlobalTracer.buildSpan(GlobalTracer.java:133)
	at io.opentracing.contrib.interceptors.OpenTracingInterceptor.wrap(OpenTracingInterceptor.java:44)
	at io.opentracing.contrib.interceptors.OpenTracingInterceptor_Bean.intercept(Unknown Source)
	at io.quarkus.arc.InvocationContextImpl$InterceptorInvocation.invoke(InvocationContextImpl.java:270)
	at io.quarkus.arc.InvocationContextImpl.invokeNext(InvocationContextImpl.java:149)
	at io.quarkus.arc.InvocationContextImpl.proceed(InvocationContextImpl.java:173)
	at io.quarkus.arc.ActivateRequestContextInterceptor.aroundInvoke(ActivateRequestContextInterceptor.java:38)
	at io.quarkus.arc.ActivateRequestContextInterceptor_Bean.intercept(Unknown Source)
	at io.quarkus.arc.InvocationContextImpl$InterceptorInvocation.invoke(InvocationContextImpl.java:270)
	at io.quarkus.arc.InvocationContextImpl.invokeNext(InvocationContextImpl.java:149)
	at io.quarkus.arc.InvocationContextImpl.proceed(InvocationContextImpl.java:173)
	at org.acme.opentracing.CounterBean_Subclass.increment(Unknown Source)
	at org.acme.opentracing.CounterBean_ClientProxy.increment(Unknown Source)
	at org.acme.opentracing.CounterBean_ScheduledInvoker_increment_1c777d19e98d20799c6b72c29b26289f3f771250.invoke(Unknown Source)
	at io.quarkus.scheduler.runtime.QuartzScheduler$InvokerJob.execute(QuartzScheduler.java:264)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	... 1 more
Caused by: java.lang.IllegalArgumentException: A metric with name jaeger_tracer_reporter_queue_length already exists
	at io.smallrye.metrics.MetricsRegistryImpl.register(MetricsRegistryImpl.java:113)
	at io.quarkus.jaeger.runtime.QuarkusJaegerMetricsFactory.createGauge(QuarkusJaegerMetricsFactory.java:64)
	at io.jaegertracing.internal.metrics.Metrics.createMetrics(Metrics.java:66)
	... 25 more

```

@pavolloffay it is still the reproducer in github that is causing the issue. Here is a full log from the reproducer:

λ mvn clean compile quarkus:dev
Java HotSpot(TM) 64-Bit Server VM warning: Ignoring option PermSize; support was removed in 8.0
[INFO] Scanning for projects...
[INFO]
[INFO] ----------------------< org.acme:getting-started >----------------------
[INFO] Building getting-started 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ getting-started ---
[INFO] Deleting C:\dev\sandbox\quarkus-scheduled-error\target
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ getting-started ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO]
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ getting-started ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to C:\dev\sandbox\quarkus-scheduled-error\target\classes
[INFO]
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:dev (default-cli) @ getting-started ---
Listening for transport dt_socket at address: 5005
2019-05-06 14:12:45,626 INFO  [io.qua.dep.QuarkusAugmentor] (main) Beginning quarkus augmentation
2019-05-06 14:12:46,545 INFO  [io.qua.dep.QuarkusAugmentor] (main) Quarkus augmentation completed in 919ms
2019-05-06 14:12:47,018 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT started in 1.515s. Listening on: http://[::]:8080
2019-05-06 14:12:47,023 INFO  [io.quarkus] (main) Installed features: [cdi, jaeger, resteasy, scheduler, smallrye-metrics, smallrye-opentracing]
2019-05-06 14:12:47,044 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-2) Register metric [name: jaeger_tracer_traces, type: counter]
2019-05-06 14:12:47,044 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-1) Register metric [name: jaeger_tracer_traces, type: counter]
2019-05-06 14:12:47,047 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-2) Register metric [name: jaeger_tracer_started_spans, type: counter]
2019-05-06 14:12:47,049 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-1) Register metric [name: jaeger_tracer_started_spans, type: counter]
2019-05-06 14:12:47,051 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-2) Register metric [name: jaeger_tracer_finished_spans, type: counter] 2019-05-06 14:12:47,053 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-1) Register metric [name: jaeger_tracer_finished_spans, type: counter] 2019-05-06 14:12:47,054 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-2) Register metric [name: jaeger_tracer_span_context_decoding_errors, type: counter]
2019-05-06 14:12:47,057 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-1) Register metric [name: jaeger_tracer_span_context_decoding_errors, type: counter]
2019-05-06 14:12:47,059 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-2) Register metric [name: jaeger_tracer_reporter_spans, type: counter] 2019-05-06 14:12:47,062 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-1) Register metric [name: jaeger_tracer_reporter_spans, type: counter] 2019-05-06 14:12:47,064 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-2) Register metric [name: jaeger_tracer_sampler_queries, type: counter]

2019-05-06 14:12:47,067 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-2) Register metric [name: jaeger_tracer_sampler_updates, type: counter]

2019-05-06 14:12:47,067 ERROR [org.qua.cor.JobRunShell] (DefaultQuartzScheduler_Worker-1) Job io.quarkus.scheduler.Scheduler.1_org.acme.quickstart.Scheduler1_ScheduledInvoker_run_72e66771a77415a7284d3ae42331659c186071de threw an unhandled Exception: : java.lang.RuntimeException: No reflection exceptions should be thrown unless there is a fundamental error in your code set up.
        at io.jaegertracing.internal.metrics.Metrics.createMetrics(Metrics.java:76)
        at io.jaegertracing.internal.metrics.Metrics.<init>(Metrics.java:32)
        at io.jaegertracing.internal.metrics.Metrics.<init>(Metrics.java:28)
        at io.jaegertracing.Configuration.getTracerBuilder(Configuration.java:221)
        at io.jaegertracing.Configuration.getTracer(Configuration.java:245)
        at io.quarkus.jaeger.runtime.QuarkusJaegerTracer.lambda$tracer$0(QuarkusJaegerTracer.java:45)
        at java.base/java.util.concurrent.atomic.AtomicReference.updateAndGet(AtomicReference.java:209)
        at io.quarkus.jaeger.runtime.QuarkusJaegerTracer.tracer(QuarkusJaegerTracer.java:41)
        at io.quarkus.jaeger.runtime.QuarkusJaegerTracer.buildSpan(QuarkusJaegerTracer.java:51)
        at io.opentracing.util.GlobalTracer.buildSpan(GlobalTracer.java:133)
        at io.quarkus.smallrye.opentracing.runtime.TracerProducer_ProducerMethod_tracer_96dadb3d6afa0cccadfe742c3e06ad433737c844_ClientProxy.buildSpan(Unknown Source)
        at io.opentracing.contrib.interceptors.OpenTracingInterceptor.wrap(OpenTracingInterceptor.java:44)
        at io.opentracing.contrib.interceptors.OpenTracingInterceptor_Bean.intercept(Unknown Source)
        at io.quarkus.arc.InvocationContextImpl$InterceptorInvocation.invoke(InvocationContextImpl.java:270)
        at io.quarkus.arc.InvocationContextImpl.invokeNext(InvocationContextImpl.java:149)
        at io.quarkus.arc.InvocationContextImpl.proceed(InvocationContextImpl.java:173)
        at io.quarkus.arc.ActivateRequestContextInterceptor.aroundInvoke(ActivateRequestContextInterceptor.java:38)
        at io.quarkus.arc.ActivateRequestContextInterceptor_Bean.intercept(Unknown Source)
        at io.quarkus.arc.InvocationContextImpl$InterceptorInvocation.invoke(InvocationContextImpl.java:270)
        at io.quarkus.arc.InvocationContextImpl.invokeNext(InvocationContextImpl.java:149)
        at io.quarkus.arc.InvocationContextImpl.proceed(InvocationContextImpl.java:173)
        at org.acme.quickstart.Scheduler1_Subclass.run(Unknown Source)
        at org.acme.quickstart.Scheduler1_ClientProxy.run(Unknown Source)
        at org.acme.quickstart.Scheduler1_ScheduledInvoker_run_72e66771a77415a7284d3ae42331659c186071de.invoke(Unknown Source)
        at io.quarkus.scheduler.runtime.QuartzScheduler$InvokerJob.execute(QuartzScheduler.java:264)
        at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
        at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: java.lang.IllegalArgumentException: A metric with name jaeger_tracer_reporter_queue_length already exists
        at io.smallrye.metrics.MetricsRegistryImpl.register(MetricsRegistryImpl.java:113)
        at io.quarkus.jaeger.runtime.QuarkusJaegerMetricsFactory.createGauge(QuarkusJaegerMetricsFactory.java:64)
        at io.jaegertracing.internal.metrics.Metrics.createMetrics(Metrics.java:66)
        ... 26 more

2019-05-06 14:12:47,071 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-2) Register metric [name: jaeger_tracer_baggage_updates, type: counter]

2019-05-06 14:12:47,074 ERROR [org.qua.cor.ErrorLogger] (DefaultQuartzScheduler_Worker-1) Job (io.quarkus.scheduler.Scheduler.1_org.acme.quickstart.Scheduler1_ScheduledInvoker_run_72e66771a77415a7284d3ae42331659c186071de threw an exception.: org.quartz.SchedulerException: Job threw an unhandled exception. [See nested exception: java.lang.RuntimeException: No reflection exceptions should be thrown unless there is a fundamental error in your code set up.]
        at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
        at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: java.lang.RuntimeException: No reflection exceptions should be thrown unless there is a fundamental error in your code set up.
        at io.jaegertracing.internal.metrics.Metrics.createMetrics(Metrics.java:76)
        at io.jaegertracing.internal.metrics.Metrics.<init>(Metrics.java:32)
        at io.jaegertracing.internal.metrics.Metrics.<init>(Metrics.java:28)
        at io.jaegertracing.Configuration.getTracerBuilder(Configuration.java:221)
        at io.jaegertracing.Configuration.getTracer(Configuration.java:245)
        at io.quarkus.jaeger.runtime.QuarkusJaegerTracer.lambda$tracer$0(QuarkusJaegerTracer.java:45)
        at java.base/java.util.concurrent.atomic.AtomicReference.updateAndGet(AtomicReference.java:209)
        at io.quarkus.jaeger.runtime.QuarkusJaegerTracer.tracer(QuarkusJaegerTracer.java:41)
        at io.quarkus.jaeger.runtime.QuarkusJaegerTracer.buildSpan(QuarkusJaegerTracer.java:51)
        at io.opentracing.util.GlobalTracer.buildSpan(GlobalTracer.java:133)
        at io.quarkus.smallrye.opentracing.runtime.TracerProducer_ProducerMethod_tracer_96dadb3d6afa0cccadfe742c3e06ad433737c844_ClientProxy.buildSpan(Unknown Source)
        at io.opentracing.contrib.interceptors.OpenTracingInterceptor.wrap(OpenTracingInterceptor.java:44)
        at io.opentracing.contrib.interceptors.OpenTracingInterceptor_Bean.intercept(Unknown Source)
        at io.quarkus.arc.InvocationContextImpl$InterceptorInvocation.invoke(InvocationContextImpl.java:270)
        at io.quarkus.arc.InvocationContextImpl.invokeNext(InvocationContextImpl.java:149)
        at io.quarkus.arc.InvocationContextImpl.proceed(InvocationContextImpl.java:173)
        at io.quarkus.arc.ActivateRequestContextInterceptor.aroundInvoke(ActivateRequestContextInterceptor.java:38)
        at io.quarkus.arc.ActivateRequestContextInterceptor_Bean.intercept(Unknown Source)
        at io.quarkus.arc.InvocationContextImpl$InterceptorInvocation.invoke(InvocationContextImpl.java:270)
        at io.quarkus.arc.InvocationContextImpl.invokeNext(InvocationContextImpl.java:149)
        at io.quarkus.arc.InvocationContextImpl.proceed(InvocationContextImpl.java:173)
        at org.acme.quickstart.Scheduler1_Subclass.run(Unknown Source)
        at org.acme.quickstart.Scheduler1_ClientProxy.run(Unknown Source)
        at org.acme.quickstart.Scheduler1_ScheduledInvoker_run_72e66771a77415a7284d3ae42331659c186071de.invoke(Unknown Source)
        at io.quarkus.scheduler.runtime.QuartzScheduler$InvokerJob.execute(QuartzScheduler.java:264)
        at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
        ... 1 more
Caused by: java.lang.IllegalArgumentException: A metric with name jaeger_tracer_reporter_queue_length already exists
        at io.smallrye.metrics.MetricsRegistryImpl.register(MetricsRegistryImpl.java:113)
        at io.quarkus.jaeger.runtime.QuarkusJaegerMetricsFactory.createGauge(QuarkusJaegerMetricsFactory.java:64)
        at io.jaegertracing.internal.metrics.Metrics.createMetrics(Metrics.java:66)
        ... 26 more

2019-05-06 14:12:47,076 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-2) Register metric [name: jaeger_tracer_baggage_truncations, type: counter]
2019-05-06 14:12:47,089 INFO  [io.sma.met.MetricsRegistryImpl] (DefaultQuartzScheduler_Worker-2) Register metric [name: jaeger_tracer_baggage_restrictions_updates, type: counter]
2019-05-06 14:12:47,139 INFO  [io.jae.Configuration] (DefaultQuartzScheduler_Worker-2) Initialized tracer=JaegerTracer(version=Java-0.34.0, serviceName=getting-started, reporter=RemoteReporter(sender=UdpSender(), closeEnqueueTimeout=1000), sampler=ConstSampler(decision=true, tags={sampler.type=const, sampler.param=true}), tags={hostname=SNBWS0027, jaeger.version=Java-0.34.0, ip=10.0.75.1}, zipkinSharedRpcSpan=false, expandExceptionLogs=false, useTraceId128Bit=false)

I was able to reproduce it. There is a race condition in https://github.com/quarkusio/quarkus/blob/991e5a31b1d4e87739c807df89accaa66f17b994/extensions/jaeger/runtime/src/main/java/io/quarkus/jaeger/runtime/QuarkusJaegerTracer.java#L41

```java
    Tracer tracer() {
        return REF.updateAndGet((orig) -> {
            if (orig != null) {
                return orig;
            }
            return Configuration.fromEnv().withMetricsFactory(new QuarkusJaegerMetricsFactory()).getTracer();
        });
    }
```
The func supplied to `updateAndGet` can be executed multiple times and the function should be "side-effect-free" which is not in this case because it creates the same metrics which is illegal hence `Caused by: java.lang.IllegalArgumentException: A metric with name jaeger_tracer_reporter_queue_length already exists` https://stackoverflow.com/a/33267333/4158442

> Atomically updates the current value with the results of applying the given function, returning the updated value. The function should be side-effect-free, since it may be re-applied when attempted updates fail due to contention among threads.

--- a/extensions/jaeger/runtime/src/main/java/io/quarkus/jaeger/runtime/QuarkusJaegerTracer.java
+++ b/extensions/jaeger/runtime/src/main/java/io/quarkus/jaeger/runtime/QuarkusJaegerTracer.java
@@ -16,8 +16,6 @@
 
 package io.quarkus.jaeger.runtime;
 
-import java.util.concurrent.atomic.AtomicReference;
-
 import io.jaegertracing.Configuration;
 import io.opentracing.ScopeManager;
 import io.opentracing.Span;
@@ -27,23 +25,23 @@ import io.opentracing.propagation.Format;
 
 public class QuarkusJaegerTracer implements Tracer {
 
-    static AtomicReference<Tracer> REF = new AtomicReference<>();
-
-    public QuarkusJaegerTracer() {
-    }
+    private static volatile Tracer tracer;
 
     @Override
     public String toString() {
         return tracer().toString();
     }
 
-    Tracer tracer() {
-        return REF.updateAndGet((orig) -> {
-            if (orig != null) {
-                return orig;
+    private static Tracer tracer() {
+        if (tracer == null) {
+            synchronized (QuarkusJaegerTracer.class) {
+                if (tracer == null) {
+                    tracer = Configuration.fromEnv()
+                            .withMetricsFactory(new QuarkusJaegerMetricsFactory()).getTracer();
+                }
             }
-            return Configuration.fromEnv().withMetricsFactory(new QuarkusJaegerMetricsFactory()).getTracer();
-        });
+        }
+        return tracer;
     }
 
     @Override

